src/proof/nova.rs

```diff
@@ -45,53 +44,38 @@ use crate::{
 /// (currently Pallas/Vesta and BN254/Grumpkin). It being pegged on the `LurkField` trait encodes that we do
 /// not expect more than one such cycle to be supported at a time for a given field.
 pub trait CurveCycleEquipped: LurkField {
-    /// ## Why the next 4 types?
-    ///
-    /// The next 4 types are purely technical, and aim at laying out type bounds in a way that rust can find them.
-    /// They should eventually be replaceable by a bound on projections, once bounds on associated types progress.
-    /// They are technically equivalent to bounds of
-    ///  <Self::G1::CE as CommitmentEngineTrait<Self::G1>>::CommitmentKey: CommitmentKeyExtTrait<Self::G1, CE = <Self::G1 as Group>::CE>,
-    ///  <Self::G2::CE as CommitmentEngineTrait<Self::G2>>::CommitmentKey: CommitmentKeyExtTrait<Self::G2, CE = <G2 as Group>::CE>,
-    /// but where clauses can't be *found* by the compiler at the point where Self::G1, Self::G2 are used
-
-    /// ## OK, but why do we need bounds at all in the first place?
+    /// ## Why the next 2 types?
+
+    /// In theory it would be sufficient to abstract over the two group types of the curve cycle, but in practice Nova is a
+    /// bit idiosyncratic in the [`nova::traits::evaluation::EvaluationEngineTrait<G>`], (PCS) it uses on these (its multilinear IPA : [`nova::provider::ipa_pc::EvaluationEngine<G>`])
+    /// *and* that implementation requires an additional trait bound `CommitmentKeyExtTrait` for this type.
```

Comment:
Is this part about `CommitmentKeyExtTrait` still applicable?

Reply:
Yes.